staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -284,10 +285,28 @@ var CommonNameUserConversion = UserConversionFunc(func(chain []*x509.Certificate
 	return &authenticator.Response{
 		User: &user.DefaultInfo{
 			Name:   chain[0].Subject.CommonName,
+			UID:    parseUIDFromCert(chain[0]),
 			Groups: chain[0].Subject.Organization,
 			Extra: map[string][]string{
 				user.CredentialIDKey: {id},
 			},
 		},
 	}, true, nil
 })
+
+// Using CNCF PEN for Kubernetes:
+// https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/4317-pod-certificates#x509-podidentity-extension
+var uidOID = asn1.ObjectIdentifier([]int{1, 3, 6, 1, 4, 1, 57683, 2})
```

Comment:
if you want other certificate signing components to re-use this it should go in e.g. k8s.io/component-base; k8s.io/apiserver doesn't seem that appropriate

Reply:
All the existing x509 client validation lives in staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go currently

This should go with that; if we intend to relocate x509 authentication, all of it should relocate together

Reply:
If other parts of the codebase want to create a client cert, they only need to rely on the std lib because of the current CN and O usage.  With this change you need to know this magic OID, so I could see some argument for putting it in k8s.io/api or similar.

Reply:
Thanks for pointing this out.

Performing the refactor in this PR seems a bit much, but I'll explore doing that next. I'll have a followup PR to use the new OID in kubeadm, which is a good opportunity to think about this re-usability issue.